blocks/text.js

```diff
@@ -779,10 +779,10 @@ Blockly.Constants.Text.TEXT_JOIN_MUTATOR_MIXIN = {
    * @this {Blockly.Block}
    */
   saveConnections: function(containerBlock) {
-    var itemBlock = containerBlock.getInputTargetBlock('STACK');
-    var i = 0;
+    let itemBlock = containerBlock.getInputTargetBlock('STACK');
+    let i = 0;
     while (itemBlock) {
```

Comment:
Possibly here too, thought the loop update condition is possibly a bit verbose unless you leave the `itemBlock` assignment in the body and only move the `i++` to the `for` line.

Reply:
I changed this one around but agree that it looks weird.

Reply:
The `no-cond-assign` rule does not like this (for readability) so I'm going to put it back. I don't think that `for (let i = 0; itemBlock; i++)` does as good a job of making it obvious that the loop depends on `itemBlock` instead of just being a standard for loop.